### PR TITLE
Implement patching remote config

### PR DESCRIPTION
**Pull request by josefkarasek**
## Description

Changes proposed in this pull request:

- Implement remote config patching suing graphql

## Testing

1. Deploy backend with instructions from cloud [PR](https://github.com/kubeshop/botkube-cloud/pull/181).

2. Run botkube `go run cmd/botkube/main.go`
```
export CONFIG_PROVIDER_IDENTIFIER="deployment-id..."
export CONFIG_PROVIDER_API_KEY="key:..."
export CONFIG_PROVIDER_ENDPOINT="http://localhost:8080/graphql"
```

4. Issue patch requests:

```
@Botkube enable notifications
@Botkube disable notifications
@Botkube status notifications
@Botkube edit sourcebindings
@Botkube status sourcebindings
```

## Related issue(s)

https://github.com/kubeshop/botkube-cloud/issues/176

### CODE REVIEW

**Review by mszostok — COMMENTED** [review]
mostly minor comments 👍

File: pkg/config/manager_remote.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	return m.gql.Cli.Mutate(ctx, &mutation, variables)
+}
+
+func (m *RemoteConfigPersistenceManager) PersistFilterEnabled(ctx context.Context, name string, enabled bool) error {
+	panic("Filter moved to kubectl plugin")
+}
+
+func (m *RemoteConfigPersistenceManager) PersistActionEnabled(ctx context.Context, name string, enabled bool) error {
+	panic("Implement me")
```
[the remark this answers is not in the document]
Comment (josefkarasek) [reply pkg/config/manager_remote.go:100]: Backend doesn't support actions yet

https://github.com/kubeshop/botkube-cloud/blob/main/botkube-cloud-backend/internal/domain/deployment/cfg_generator.go#L47

File: internal/remote/api.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package remote
```
Comment (josefkarasek) [inline internal/remote/api.go:1]: @mszostok any better better place for remote API objects?

File: pkg/config/manager_remote.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "while persisting source bindings config")
+	}
+	return nil
+}
+
+func (m *RemotePersistenceManager) PersistActionEnabled(ctx context.Context, name string, enabled bool) error {
+	return errors.New("PersistActionEnabled is not implemented for GQL manager")
```
Comment (mszostok) [inline pkg/config/manager_remote.go:136]: do we already have a follow-up task?

**Review by mszostok — APPROVED** [review]
Great work 👍 Please only fix test issues before merging.

**Pull request merged**